runtime/sema/gen/main.go

```diff
@@ -164,6 +164,9 @@ type typeDecl struct {
 	memberDeclarations []ast.Declaration
 	nestedTypes        []*typeDecl
 	hasConstructor     bool
+
+	// used in simpleType generation
+	conformances []string
```

Comment:
Maybe consider making this more type-safe by using a `[]common.TypeID` instead

Reply:
Changed to an `InterfaceType` for simplicity